code/scripting/api/objs/player.cpp

```diff
@@ -700,6 +764,39 @@ ADE_FUNC(getMissionShipclassKills,
 
 	return ade_set_args(L, "i", ssh->get()->m_kills[ship_idx]);
 }
+
+ADE_FUNC(setMissionShipclassKills,
+	l_ScoringStats,
+	"shipclass class, number kills",
+	"Sets the number of kills of a specific ship class recorded in this statistics structure for the current "
+	"mission. Returns true if successful.",
+	"boolean",
+	"True if successful")
+{
+	using namespace luacpp;
+
+	scoring_stats_h* ssh;
+	int ship_idx;
+	int setv;
+	if (!ade_get_args(L, "ooi", l_ScoringStats.GetPtr(&ssh), l_Shipclass.Get(&ship_idx), &setv)) {
+		return ade_set_error(L, "i", -1);
+	}
+
+	if (!ssh->isValid()) {
+		return ade_set_error(L, "i", -1);
+	}
+
+	if (ship_idx < 0 || ship_idx >= (int)Ship_info.size()) {
+		return ade_set_error(L, "i", -1);
```

Comment:
These three should probably return false and not -1, based on your documentation

Reply:
Good call. Done.